draft-ietf-tls-esni.md

```diff
@@ -277,7 +277,16 @@ of a particular ECH configuration.
 public_name
 : The non-empty name of the client-facing server, i.e., the entity trusted to
 update the ECH configuration. This is used to correct misconfigured clients, as
-described in {{handle-server-response}}.
+described in {{handle-server-response}}. Clients MUST ignore any `ECHConfig`
+structure with a `public_name` that begins or ends with an ASCII dot or cannot
+be parsed as a dot-separated sequence of LDH labels, as defined in {{RFC5890}},
+Section 2.3.1. This validation procedure is incomplete; it incidentally rules
```

Comment:
Perhaps rephrasing this slightly as:

> This value MUST NOT begin or end with an ASCII dot and MUST be parsable as a dot-separated sequence of LDH labels, as defined in {{RFC5980}}, Section 2.3.1. Clients MUST ignore any ECHConfig structure whose public_name does not meet this criteria.

This way we capture both sender and receiver rules.

Reply:
That's better, thanks.